### PR TITLE
cloudapi: add support for multi-tenancy

### PR DESCRIPTION
**Pull request by ondrejbudai**
This PR adds support for multi-tenancy. :partying_face: 

Currently, only `PostCompose` in Cloud API and `RequestJob` in Remote worker API are tenant-aware. Also, multi-tenancy is only enabled when JWT auth is on. For other use-cases (mTLS in cloud and koji API & weldr), the channel name (more on that later) is set to an empty string which basically means there's no tenant. We should probably expand tenant-awareness to all routes but it's not needed for the MVP.

## Big picture
The route handlers extract `account_id`, or `rh-org-id` (configurable) from JWT claims. These are IDs of organizations on access.redhat.com. This means that access.redhat.com organization == Image Builder tenant.

The IDs are then attached to each job belonging to the tenant in the form of a channel (with the prefix `org-`). Channels serve a similar purpose in the jobqueue as job types: When enqueuing a job, a channel must be specified. When dequeueing a job, a list of accepted channels must be specified.

---
 
You are encouraged to review this PR commit by commit, these are some of the important remarks:

- There's a DB migration involved as we need a new `channel` column in `jobs` table. Also, the view must be recreated to contain the new column.
- I introduced Config structs for cloud and worker API servers to reduce the number of arguments. TBH, I would love us to start doing this more.
- `osbuild-mock-openid-provider` now uses the refresh token as rh-org-id claim in the JWT. Hacky but it works quite nicely for our tests.
- I started restructuring composer's config a bit. The `koji` section is meant to die in the near future.
- I started writing a small library for writing API tests in Go. See the commit message. tl;dr: `TestRoute` isn't flexible enough and it's extremely annoying to add new features to it.
- `tenant_provider_fields` were added to the api.openshift.com config so once this commit is merged, we should just have multi-tenancy working on stage. :champagne:

## Tenants & channels thoughts

I decided to split the concept of tenants and channels. You may want to know why:

We may want to share channels between tenants in the future. E.g. both copr and koji want to build Fedora images on a shared pool of Fedora workers.

With my design, it's easy to add a channel argument to both `PostComposer` and `RequestJob` calls that can override the default `org-${tenant}` channel assigned to a tenant. Also, composer would have to have some kind of ACL. e.g. copr and fedora tenant can access fedora-workers channel. And that's basically it. Multiple tenants can share a channel with very small amount of work on our side.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Comment by gicmo** [comment]
@ondrejbudai oh, RH infra supports a refresh token? Last time I poked it seems to not have one. I guess I should update koji-osbuild accordingly.

**Review by Gundersanne — COMMENTED** [review]
Amazing, just 1 thought.

**Comment by Gundersanne** [comment]
> @ondrejbudai oh, RH infra supports a refresh token? Last time I poked it seems to not have one. I guess I should update koji-osbuild accordingly.

@gicmo Not sure if koji-osbuild needs to support it, as that will use the clientid/secret (being the identity.api.openshift.com service accounts).

File: cmd/osbuild-composer/config.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 
 type ComposerConfigFile struct {
 	Koji      KojiAPIConfig   `toml:"koji"`
+	CloudAPI  CloudAPIConfig  `toml:"cloud_api"`
```
[the remark this answers is not in the document]
Comment (ondrejbudai) [reply cmd/osbuild-composer/config.go]: Agreed, I simplified the config changes to a bare minimum. If we ever restructure the config, we should do it in a follow-up.

**Comment by ondrejbudai** [comment]
Force-pushed a new version that simplifies the new config option for `tenant_provider_fields`.

**Review by Gundersanne — APPROVED** [review]
LGTM!

**Comment by ondrejbudai** [comment]
I forgot to also change the openshift template when changing the config keys, fixed.

**Review by Gundersanne — APPROVED** [review]
LgooderTM

**Review by teg — APPROVED** [review]
This is great! I'm not 100% sure of the future extensions to the channels, but now they are 1to1 with tenants, so that's something we could discuss later.

Another possible future improvement is to guard everything behind tenant ID/channel, not only the two API calls you did so far (but it isn't necessary afaict).

**Pull request merged**